### PR TITLE
Allow a default config to be used 

### PR DESCRIPTION
**Pull request by academo**
This PR's adds a default config to the plugin and an option to enable it when you init.

The default config is meant to ease the setup of diagnostics and formatting for the same standard files that are supported by the plugin.

To avoid breaking any configuration from people using the plugin, the default configuration is opt-in.

To enable this default configuration a new parameter can be passed to the init function

`require'diagnosticls-nvim'.init {default_config = true}`

This will internally set all the linters and formatters.

A user could change or disable specific linters or formatters by passing custom config. E.g. If eslint would like to be used to fix JS files instead of prettier, an user could write this:

```lua
require'diagnosticls-nvim'.init {default_config = true}

require'diagnosticls-nvim'.setup {
    ['javascript'] = {
		linter = require 'diagnosticls-nvim.linters.eslint',
        formatter = require 'diagnosticls-nvim.formatters.eslint'
    },
    
}
```

note that in the previous snippet the linter is also passed. If not passed the linter will be empty: the default config containing linter and formatter as keys will be overwritten by a new table containing only formatter. 

This PR also adds an additional option: `format`. 

The idea is, if some use wants to use the default_config for linting for not for formatting, they could pass this to the init and the plugin will not set any formatter (not even the ones passed on the setup function).

This is useful for users that'd like to use this plugin only for linting with the default config.

Note: I applied lua-format to the init.lua file.

### CODE REVIEW

**Comment by creativenull** [comment]
Hey, thanks for the PR 🙂

I'll take a look at it during this week and see if it's good or if it needs some improvements

File: README.md [hunk 1]
```diff
@@ -7,9 +7,9 @@ Have a look at the currently [supported linters/formatters](#supported-linters-a
 
 ### Requirements
 
-+ Neovim v0.5
-+ DiagnosticLS, globally installed: `npm i -g diagnostic-languageserver`
-+ `nvim-lspconfig`
+- Neovim v0.5
+- DiagnosticLS, globally installed: `npm i -g diagnostic-languageserver`
+- `nvim-lspconfig`
```
Comment (creativenull) [inline README.md:12]: Don't think it's needed to change the formatting of the list

File: README.md [hunk 2]
```diff
@@ -53,52 +53,73 @@ require 'diagnosticls-nvim'.setup {
...
+  - [`reek`][reek] => `require 'diagnosticls-nvim.linters.reek'` --default
+  - [`rubocop`][rubocop] => `require 'diagnosticls-nvim.linters.rubocop'`
+- CSS/SCSS/LESS
+  - [`stylelint`][stylelint] => `require 'diagnosticls-nvim.linters.stylelint'` --default
+- PHP
+  - [`phpcs`][phpcs] => `require 'diagnosticls-nvim.linters.phpcs'` --default
+  - [`phpstan`][phpstan] => `require 'diagnosticls-nvim.linters.phpstan'`
+  - [`psalm`][psalm] => `require 'diagnosticls-nvim.linters.psalm'`
+- Vim
+  - [`vint`][vint] => `require 'diagnosticls-nvim.linters.vint'` --default
```
Comment (creativenull) [inline README.md:122]: Same here

File: lua/diagnosticls-nvim/formatters/eslint.lua [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  args = { '--fix', '--fix-to-stdout', '--stdin' '--stdin-filename', '%filepath' },
+  rootPatterns = {
+    '.eslintrc',
+    '.eslintrc.cjs',
+    '.eslintrc.js',
+    '.eslintrc.json',
+    '.eslintrc.yaml',
+    '.eslintrc.yml',
+  },
+}
```
Comment (creativenull) [inline lua/diagnosticls-nvim/formatters/eslint.lua:15]: I'd rather this be another PR instead, to add the linter, and have this PR focus on the default configs

File: lua/diagnosticls-nvim/init.lua [hunk 4]
```diff
@@ -2,59 +2,97 @@ local lspconfig = require 'lspconfig'
 local M = {}
 
 local diagnosticls_nvim_lsp_opts = {
-  root_dir = require 'lspconfig'.util.root_pattern('.git')
```
Comment (creativenull) [inline lua/diagnosticls-nvim/init.lua:5]: I wanna keep the line indents to 2 spaces for now, I should add an editorconfig to normalize the indents

**Comment by academo** [comment]
@creativenull any comment on the PR other than code formatting/style?

**Comment by creativenull** [comment]
From what I see the code looks good to add default configs to the `setup()` function, I've also tested it out and it seems to work. I also feel like the `format` key option should also be a different PR but it's fine to keep it here.

Let me add an editorconfig and lua format rules to the project and let you know when ready for you to make the formatting changes. In the meantime, you can move the eslint formatter config to a different PR.

**Comment by academo** [comment]
Great! I'll move eslint fixer to another PR.

could you please ping me when you have the editorconfig? thanks!

**Comment by creativenull** [comment]
@academo I've added a editorconfig and lua-format rules to the repo. But I can fix those formatting issue myself and go ahead and approve this if you're good

**Comment by academo** [comment]
@creativenull go for it! thanks
